File: ios/Classes/AlbumConversions.m
//
//  AlbumConversions.h
//  album_manager
//
//  Created by LeeWei on 2022/8/11.
//

#import "AlbumConversions.h"
#import "AlbumAsset.h"

@implementation AlbumConversions

+(NSDictionary*) convertAlbumAssetToMap:(AlbumAsset *)asset
{
    return @{
        @"id" : asset.id,
        @"title" : asset.title,
        @"mimeType" : asset.mimeType,
        @"creationDate" : @(asset.creationDate),
        @"modifiedDate" : @(asset.modifiedDate),
        @"lng" : @(asset.lng),
        @"lat" : @(asset.lat),
    };
}

+(NSDictionary*) convertAssetMetaDictToMap:(NSDictionary *)meta {
    return @{
        @"ColorModel": [NSString stringWithFormat:@"%@", [meta valueForKey:@"ColorModel"]],
        @"DPIHeight": @([[NSString stringWithFormat:@"%@", [meta valueForKey:@"DPIHeight"]]  doubleValue]),
        @"DPIWidth": @([[NSString stringWithFormat:@"%@", [meta valueForKey:@"DPIWidth"] ] doubleValue]),
        @"Depth": @([[NSString stringWithFormat:@"%@", [meta valueForKey:@"Depth"] ] intValue]),
        @"PixelHeight": @([[NSString stringWithFormat:@"%@", [meta valueForKey:@"PixelHeight"] ] intValue]),
        @"PixelWidth": @([[NSString stringWithFormat:@"%@", [meta valueForKey:@"PixelWidth"] ] intValue]),
        @"ProfileName": [NSString stringWithFormat:@"%@", [meta valueForKey:@"ColorModel"]],
//        @"Exif" : [meta objectForKey:@"{Exif}"],
//        @"TIFF" : [meta objectForKey:@"{TIFF}"],
//        @"GPS" : [meta objectForKey:@"{GPS}"],
    };
}

-(NSString*) strinVauleForKey: id
{
    return [NSString stringWithFormat:@"%@", id];
}

-(NSInteger) intVauleForKey: id
{
    return [[NSString stringWithFormat:@"%@", id] integerValue];
}
@end

File: lib/album_conversions.dart
import 'package:album_manager/types/album_asset.dart';
import 'package:album_manager/types/asset_meta.dart';

class AlbumConversions {
  static AlbumAsset convertMapToAlbumAsset(Map<String, dynamic> data) {
    return AlbumAsset(
      id: data['id'] as String,
      title: data['title'] as String,
      mimeType: data['mimeType'] as String,
      creationDate: data['creationDate'] as int?,
      modifiedDate: data['modifiedDate'] as int?,
      latitude: data['lat'] as double?,
      longitude: data['lng'] as double?,
    );
  }

  static AssetMeta convertMapToAssetMeta(Map<String, dynamic> data) {
    return AssetMeta(
      colorModel: data['ColorModel'] as String,
      dpiWidth: data['DPIWidth'] as double,
      dpiHeight: data['DPIHeight'] as double,
      depth: data['Depth'] as int,
      pixelHeight: data['PixelHeight'] as int,
      pixelWidth: data['PixelWidth'] as int,
      profileName: data['ProfileName'] as String,
    );
  }
}
